nemo_skills/evaluation/metrics/ioi_metrics.py

```diff
@@ -0,0 +1,106 @@
+from collections import defaultdict
+
+
+class IOIMetrics(BaseMetrics):
+    def __init__(self):
+        super().__init__()
+        self.reset()
+
+    def update(self, predictions):
+        if len(predictions) > 1:
+            self.max_k = len(predictions)
+            self.agg_mode = f"pass@{len(predictions)}"
+        else:
+            self.max_k = 0
+            self.agg_mode = "greedy"
+        for pred in predictions:
+            problem_name = pred["name"]
+            self.predictions_by_problem[problem_name].append(pred)
+
+    def get_problem_score(self, submissions) -> float:
+        """
+        For a given problem (list of submissions), compute the score as follows:
+          - For each subtask, take the maximum score over all submissions.
+          - Sum these maximum scores to get the problem score.
+        """
+        if not submissions:
+            return 0.0
+        subtasks = list(submissions[0]["test_case_results"].keys())
+        subtask_scores = {subtask: 0 for subtask in subtasks}
+
+        for submission in submissions:
+            test_case_results = submission["test_case_results"]
+            for subtask, result in test_case_results.items():
+                subtask_scores[subtask] = max(subtask_scores[subtask], result["score"])
+        return sum(subtask_scores.values()), subtask_scores
+
+    def simulate_round_robin_score(self, submissions) -> float:
+        """
+        Computes a round robin score for a problem.
+        The procedure is as follows:
+         1. For each submission, compute an aggregate score (sum of subtask scores).
+         2. Sort submissions in descending order by the aggregate score.
+         3. Select up to 50 submissions.
+         4. For each subtask, take the maximum score among the selected submissions.
+         5. Return the sum of these maximum subtask scores.
+        """
+        if not submissions:
+            return 0.0
+
+        # compute an aggregate score per submission
+        for submission in submissions:
+            aggregate_score = sum(result["score"] for result in submission["test_case_results"].values())
+            submission["_aggregate_score"] = aggregate_score
+
+        # sort submissions in descending order by aggregate score
+        sorted_submissions = sorted(submissions, key=lambda s: s["_aggregate_score"], reverse=True)
+        # Select up to 50 submissions.
+        selected = sorted_submissions[:50]
+
+        # for each subtask, take the maximum score among the selected submissions
+        subtasks = list(submissions[0]["test_case_results"].keys())
+        subtask_scores = {subtask: 0 for subtask in subtasks}
+        for submission in selected:
+            for subtask, result in submission["test_case_results"].items():
+                subtask_scores[subtask] = max(subtask_scores[subtask], result["score"])
+        return sum(subtask_scores.values())
+
+    def get_metrics(self):
```

Comment:
would it be better to use a standard interface for pass@k computation? You can look at e.g. lcb or evalplus metrics for an example. Or is it too hard in this case?

Reply:
sorry, I meant using the parent class's `_compute_pass_at_k` function. You'd need to define `_get_score_dict` for that and it should work automatically after that. Not sure if this logic fits naturally in that interface, but if it does, would be great to reuse it. Have a look at other metric classes for examples